### PR TITLE
[UPDATED] - MR:completed sigup and login using ajax and drf

### PR DESCRIPTION
**Pull request by Jinal-81**

### CODE REVIEW

File: static/js/script.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            if (data.status==400 && data.responseJSON) {
+                $.each(data_response, function( v, k ) {
+                        debugger
+                        var field = $(current_form).find('#'+v+'_id');
+                        field.addClass('inputTxtError').before('<div class="error">'+k+'</div>');
+                });
+        }
+        }
+
+      var formData = {
```
Comment (abhishek145607) [inline static/js/script.js]: Use serializeArray instead of this

File: static/js/script.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        error: function (data){
+            if (callbackFailure){
+                callbackFailure(data);
+            }
+        }
+    })
+}
+// user registration
+$(document).ready(function (){
+    // id of the form for the submit
```
Comment (rahultrootech) [inline static/js/script.js:32]: multiple times $(document).ready(function (){ used in single file. move all DOM loaded content in one

File: static/js/script.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
+function ajaxPostCallRequest(url, data, callbackSuccess=null, callbackFailure=null){
+    $.ajax({
```
Comment (rahultrootech) [inline static/js/script.js:9]: create one function which will be called based on method like

function  ajaxCallRequest(method, ........){}


function getAjaxCallRequest(){
           return ajaxCallRequest("GET")
}


function postAjaxCallRequest(){
           return ajaxCallRequest("POST")
}

File: static/js/script.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                $.each(data_response, function( v, k ) {
+                        var field = $(current_form).find('#'+v+'_id');
+                        field.addClass('inputTxtError').before('<div class="error">'+k+'</div>');
+                });
+        }
+        }
+
+      var formData = {
+      username: $("#username_id").val(),
+      password: $("#password_id").val(),
```
Comment (rahultrootech) [inline static/js/script.js]: get fields value from form directly using serializerArray

File: user/api/serializers.py [hunk 5]
```diff
@@ -46,14 +46,13 @@ class Meta:
         }
 
     def validate_password(self, attrs):
-        import pdb; pdb.set_trace()
```
Comment (rahultrootech) [inline user/api/serializers.py:49]: breakpoint()

File: static/js/script.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+                    toastr.error(m.message)
+                }
+            });
+            var token = window.localStorage.getItem('accessToken');
+            var payload = JSON.parse(atob(token.split('.')[1]));
+            var username = payload.username;
+           window.location.href = current_url + '/user/dashboard/';
+           event.preventDefault(); // avoid to execute the actual submit of the form.
+           document.getElementById('username').innerText = username;
+        }
```
Comment (rahultrootech) [inline static/js/script.js:86]: use either jquery or Javascript:  document.getElementById('username').innerText

File: static/js/script.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+                    toastr.success(m.message)
+                }
+                else if (m.extra_tags=='error'){
+                    toastr.error(m.message)
+                }
+            });
+            var token = window.localStorage.getItem('accessToken');
+            var payload = JSON.parse(atob(token.split('.')[1]));
+            var username = payload.username;
+           window.location.href = current_url + '/user/dashboard/';
```
Comment (rahultrootech) [inline static/js/script.js]: can be created Mapper for url endpoint '/user/dashboard/'

File: static/js/script.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    // id of the form for the submit
+    $('#sign_form_id').submit(function (event){
+      event.preventDefault(); // avoid to execute the actual submit of the form.
+      var current_url = window.location.origin;
+      function successRegistrationThread(data){
+           console.log(data)
+
+           $.each(data.messages, function (i, m) {
+                if (m.extra_tags=='success'){
+                    toastr.success(m.message)
```
Comment (rahultrootech) [inline static/js/script.js:41]: code duplicate, same things are using below
 if (m.extra_tags=='success'){
                    toastr.success(m.message)

**Pull request merged**